js/main.js

```diff
@@ -0,0 +1,177 @@
+/* global google */
+
+/**
+ * Tag constructor
+ *
+ * @constructor
+ * @param {string} searchTerm The tag to search for
+ */
+var Tag = function(searchTerm) {
+  this.searchTerm = searchTerm;
+};
+
+/**
+ * Get the top places where this tag was used
+ *
+ * @param {function} callback An optional method to call upon completion
+ */
+Tag.prototype.getPlaces = function(callback) {
+  $.ajax({
+    url: "http://api.flickr.com/services/rest/?jsoncallback=?",
+    dataType: 'jsonp',
+    data: {
+      method: 'flickr.places.placesForTags',
+      api_key: 'b3a7fa326b8255e6af7900b7a58841e0',
+      format: 'json',
+      tags: this.searchTerm,
+      place_type_id: '22'
+    },
+    success: function(data) {
+      if (typeof callback !== 'undefined') {
+        callback(data.places.place);
+      }
+    }
+  });
+};
+
+/**
+ * Get the top photo for a given location
+ *
+ * @param {number} latitude The latitude to find the photo from
+ * @param {number} longitude The longitude to find the photo from
+ * @param {function} map The HeatMap instance to add the pictures to
+ */
+Tag.prototype.getPhotos = function(latitude, longitude, map) {
+  $.ajax({
+    url: "http://api.flickr.com/services/rest/?jsoncallback=?",
+    dataType: 'jsonp',
+    data: {
+      method: 'flickr.photos.search',
+      api_key: 'b3a7fa326b8255e6af7900b7a58841e0',
+      format: 'json',
+      tags: this.searchTerm,
+      per_page: 1,
+      lat: latitude,
+      lon: longitude,
+      extras: 'url_sq'
+    },
+    success: function(data) {
+      if (data.photos.photo.length) {
+        var src = data.photos.photo[0].url_sq;
+        var contentString = '<img src="' + src + '" alt="">';
+        var infowindow = new google.maps.InfoWindow({
+            content: contentString
+        });
+
+        var marker = new google.maps.Marker({
+            position: new google.maps.LatLng(latitude, longitude),
+            map: map,
+        });
+        google.maps.event.addListener(marker, 'click', function() {
+          infowindow.open(map, marker);
+        });
+      }
```

Comment:
When you have this much logic in the callback, you might want to move it out to a different method.